base/system/winlogon/sas.c

```diff
@@ -347,7 +361,9 @@ PlayLogonSoundThread(
     }
     else
     {
-        PlaySoundRoutine(L"WindowsLogon", TRUE, SND_ALIAS | SND_NODEFAULT);
+        PlaySoundRoutine(IsFirstLogon() ? L"SystemStart" : L"WindowsLogon",
```

Comment:
What happens if the autoadminlogin feature is disabled, and by default when ReactOS is started, you first land on the login dialog? (I also wonder about the same for windows XP, 2003 or 7)

(You would need to change the `REG_DWORD` value named `AutoAdminLogon` in `HKLM\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon` to `0` then do a full shutdown first, before testing the whole thing.)

Reply:
I will try that.

Reply:
The startup sound still works on the first logon, after signed in using login dialog. And the logon and logoff sounds behavior doesn't change.